pkg/controller/shared_state.go

```diff
@@ -1788,35 +1788,28 @@ func (c *SharedState) deleteNodeFromAvailableAttachmentsMap(ctx context.Context,
 	c.availableAttachmentsMap.Delete(node)
 }
 
-func (c *SharedState) decrementAttachmentCount(ctx context.Context, node string) bool {
+func (c *SharedState) decrementNodeCapacity(ctx context.Context, node string) bool {
 	remainingCapacity, nodeExists := c.availableAttachmentsMap.Load(node)
 	if nodeExists && remainingCapacity != nil {
-		currentCapacity := int32(0)
-		for {
-			currentCapacity = remainingCapacity.(*atomic.Int32).Load()
-			if currentCapacity == int32(0) || remainingCapacity.(*atomic.Int32).CompareAndSwap(currentCapacity, currentCapacity-1) {
-				if currentCapacity == int32(0) {
-					klog.Errorf("Failed to decrement attachment count for node(%s), because no available attachment", node)
-					return false
-				}
-				return true
-			}
+		if currentCapacity := remainingCapacity.(*atomic.Int32).Load(); currentCapacity == int32(0) {
+			klog.Errorf("Failed to decrement disk capacity for node(%s) because no remaining capacity", node)
+			return false
 		}
+		remainingCapacity.(*atomic.Int32).Add(-1)
```

Comment:
The value of `remainingCapacity` is loaded in line 1792 and gets updated in this line 1798. However, if another thread modifies the value of `remainingCapacity` in the meantime, the number will be inaccurate.